### ISSUE TITLE
Reduce queries for comments on article show page.

### ISSUE THREAD

**Issue by benhalpern** [thepracticaldev/dev.to#1606]
**Describe the bug**
Our worst-case scenario on article show pages is an n+1 problem. We do a fair amount of caching and we have a _somewhat_ efficient approach to not make _literally_ every query. But still, we do extra queries on some requests, and when a page with a lot of comments gets loaded cold, it can be a big outlier in performance.

**To Reproduce**
Create an article in development and add a bunch of comments at different levels. You should see lots of queries happen.

**Expected behavior**
A single query should be enough to load all the comments into memory. All comments that belong to an article have a `commentable_id` of that article, even sub-comments. We use the [ancestry gem](https://github.com/stefankroes/ancestry) to handle the tree logic.

Since we can grab all the comments that would be part of the tree with a single query, we _should_ be able to only ever make a single comments query and then work on that set from there.

I think the solution would be something along these lines:

https://stackoverflow.com/questions/32207398/how-to-perform-eager-loading-on-model-descendants-using-rails-and-ancestry

Thanks!

**Comment by arku** [comment]
@benhalpern I would like to take a stab at this one!

**Comment by lightalloy** [comment]
I've started work on this one.
When loading comments with one request it's harder query a limited amout of root comments + their ancestors. So I was concerned with the memory usage while arranging the whole tree (+ regarding the `includes(:user)`).
I've checked the memory usage by `memory-profiler` and it seems like there's no dramatic change comparing with the old version. With limited comments (on the article show page) the usage is even less.
However, I'll explore the options to refactor so that only the limited comments are loaded and see if there are any benefits.
I also plan to refactor the code which loads comments and displays the tree.
